From 7a57053ed7c9fa767596b33521545c552be89dba Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:08:43 +0200
Subject: [PATCH] tests: py: complete icmp and icmpv6 update

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 1702bfd70b39fe955d5e2c4b93f7dc07acb79027

commit 1702bfd70b39fe955d5e2c4b93f7dc07acb79027
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Thu Apr 4 13:56:39 2024 +0200

    tests: py: complete icmp and icmpv6 update

    Update json update and leftover payload update to complete
    5fecd2a6ef61 ("src: disentangle ICMP code types").

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 tests/py/ip/icmp.t.json           | 6 +++---
 tests/py/ip/icmp.t.json.output    | 4 ++--
 tests/py/ip6/icmpv6.t.json        | 8 ++++----
 tests/py/ip6/icmpv6.t.json.output | 8 ++++----
 tests/py/ip6/icmpv6.t.payload.ip6 | 2 +-
 5 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/tests/py/ip/icmp.t.json b/tests/py/ip/icmp.t.json
index 4f05250..45e04c7 100644
--- a/tests/py/ip/icmp.t.json
+++ b/tests/py/ip/icmp.t.json
@@ -459,8 +459,8 @@
             "op": "!=",
             "right": {
                 "set": [
-                    "prot-unreachable",
-                    "frag-needed",
+                    2,
+                    4,
                     33,
                     54,
                     56
@@ -1488,7 +1488,7 @@
                 }
             },
             "op": "==",
-            "right": "host-unreachable"
+            "right": 1
         }
     }
 ]
diff --git a/tests/py/ip/icmp.t.json.output b/tests/py/ip/icmp.t.json.output
index 5a07585..52fd601 100644
--- a/tests/py/ip/icmp.t.json.output
+++ b/tests/py/ip/icmp.t.json.output
@@ -11,8 +11,8 @@
             "op": "==",
             "right": {
                 "set": [
-                    "prot-unreachable",
-                    "frag-needed",
+                    2,
+                    4,
                     33,
                     54,
                     56
diff --git a/tests/py/ip6/icmpv6.t.json b/tests/py/ip6/icmpv6.t.json
index 224a8e8..9df886d 100644
--- a/tests/py/ip6/icmpv6.t.json
+++ b/tests/py/ip6/icmpv6.t.json
@@ -532,8 +532,8 @@
             "op": "!=",
             "right": {
                 "set": [
-                    "policy-fail",
-                    "reject-route",
+                    5,
+                    6,
                     7
                 ]
             }
@@ -1136,7 +1136,7 @@
     }
 ]
 
-# icmpv6 type parameter-problem icmpv6 code no-route
+# icmpv6 type parameter-problem icmpv6 code 0
 [
     {
         "match": {
@@ -1159,7 +1159,7 @@
                 }
             },
             "op": "==",
-            "right": "no-route"
+            "right": 0
         }
     }
 ]
diff --git a/tests/py/ip6/icmpv6.t.json.output b/tests/py/ip6/icmpv6.t.json.output
index 7b8f5c1..f29b346 100644
--- a/tests/py/ip6/icmpv6.t.json.output
+++ b/tests/py/ip6/icmpv6.t.json.output
@@ -104,7 +104,7 @@
                 }
             },
             "op": "==",
-            "right": "port-unreachable"
+            "right": 4
         }
     }
 ]
@@ -122,7 +122,7 @@
             "op": "==",
             "right": {
                 "range": [
-                    "addr-unreachable",
+                    3,
                     66
                 ]
             }
@@ -143,8 +143,8 @@
             "op": "==",
             "right": {
                 "set": [
-                    "policy-fail",
-                    "reject-route",
+                    5,
+                    6,
                     7
                 ]
             }
diff --git a/tests/py/ip6/icmpv6.t.payload.ip6 b/tests/py/ip6/icmpv6.t.payload.ip6
index fcaf481..5b6035d 100644
--- a/tests/py/ip6/icmpv6.t.payload.ip6
+++ b/tests/py/ip6/icmpv6.t.payload.ip6
@@ -554,7 +554,7 @@ ip6 test-ip6 input
   [ payload load 2b @ transport header + 4 => reg 1 ]
   [ lookup reg 1 set __set%d 0x1 ]
 
-# icmpv6 type parameter-problem icmpv6 code no-route
+# icmpv6 type parameter-problem icmpv6 code 0
 ip6 
   [ meta load l4proto => reg 1 ]
   [ cmp eq reg 1 0x0000003a ]
